### PR TITLE
Handle tx fails/errors

### LINKED ISSUE #100: Endless spinner on txn error with no escape
image

- [ ] Add guidance copy if wallet is stuck
- [ ] Restyling buttons on tx modal to match app (note the ones when txs are canceled)
- [ ] Inform user that they still need to allocate votes if 3rd tx was canceled

### PR DESCRIPTION
**Pull request by pettinarip**
Fix #100 

- Handle unexpected errors while executing the `contribute` function.
- Handle excessive tx times displaying a warning to the user.
- Fix styles in the contribution modal buttons.

### CODE REVIEW

File: vue-app/src/components/ContributionModal.vue [hunk 1]
```diff
@@ -166,112 +167,123 @@ export default class ContributionModal extends Vue {
...
+      } catch (error) {
+        this.voteTxError = error.message
+        return
+      }
+      this.step += 1
+    } catch (err) {
+      /* eslint-disable-next-line no-console */
+      console.log(err)
+      this.error =
+        'Something unexpected ocurred. Refresh the page and try again.'
```
Comment (pettinarip) [inline vue-app/src/components/ContributionModal.vue:268]: Seems like a lot of changes but I've only added a try/catch block wrapping the entire `contribute` function.

Is for handling unexpected errors. Things that are not being caught by the others try/catch.

For example, I was getting an internal RPC error and those kinds of cases were not being caught by anything, so the modal was breaking.

Comment (pettinarip) [reply vue-app/src/components/ContributionModal.vue:268]: Any suggestion about the text message is appreciated :)

File: vue-app/src/components/ContributionModal.vue [hunk 2]
```diff
@@ -285,7 +297,7 @@ export default class ContributionModal extends Vue {
   margin: 1rem 0;
   margin-top: 1.5rem;
   align-items: center;
-  justify-content: center;
+  justify-content: space-between;
```
Comment (pettinarip) [inline vue-app/src/components/ContributionModal.vue:283]: This is to keep the buttons on each side, not centered, keeping it consistent with the other modals.

**Review by samajammin — APPROVED** [review]
Code LGTM but any sense of how I could try to reproduce this error scenario?

**Comment by pettinarip** [comment]
@samajammin 
For the "unexpected errors" case, I don't have an easy way to test it. Perhaps you can just make the code fails by adding a `throw new Error()` in the middle of the `contribute` function 🤷🏼 

For the warning text, you can do this:
1. Go and contribute something
2. When the modal appears, just wait. You will need to wait for 30s for it to appear
3. Once you see the message, you can approve the tx and go to the next step
4. You can test again waiting for 30s to see the warning message again
5. And so on

I put 30 seconds. We could change it if you consider it is too much

**Comment by samajammin** [comment]
@pettinarip this looks good to me! Can you please resolve conflicts & merge? Thanks 😄

**Pull request merged**